core/search/granular_range_filter.cpp

```diff
@@ -334,11 +333,11 @@ void collect_terms_within(
   // ...........................................................................
 
   if (min_term_inclusive && !min_term.empty()) {
-    auto& exact_min_term = min_term.begin()->second;
-    bool single_term = !max_term.empty() && exact_min_term == max_term.begin()->second;
+    auto& exact_min_term = min_term.front();
+    bool single_term = !max_term.empty() && exact_min_term == max_term.front();
 
     if ((!single_term || max_term_inclusive)
-        && exact_min_term > max_term.begin()->second) {
+        && exact_min_term > max_term.front()) {
```

Comment:
Could max_term be empty there ? Than single_term is false and max_term.front() will trigger UB.
Suggest to add max_term.empty() check